tests/functional/steps/given.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+from pytest_bdd import given, parsers
+from requests import post
+from tests.functional.data import searches
+from tests.functional.data import patients
+from tests.functional.data.patients import Patient
+from tests.functional.data.searches import Search
+from tests.functional.data.updates import Update
+from tests.functional.steps.when import retrieve_patient
+from pytest_nhsd_apim.apigee_apis import ApiProductsAPI
+import json
+import time
+
+
+@given('scope added to product')
+def add_scope_to_products_patient_access(products_api: ApiProductsAPI,
+                                         nhsd_apim_proxy_name: str,
+                                         nhsd_apim_authorization: dict):
+    product_name = nhsd_apim_proxy_name.replace("-asid-required", "")
+
+    default_product = products_api.get_product_by_name(product_name=product_name)
+    if nhsd_apim_authorization['scope'] not in default_product['scopes']:
+        default_product['scopes'].append(nhsd_apim_authorization['scope'])
+        products_api.put_product_by_name(product_name=product_name, body=default_product)
+        time.sleep(2)
```

Comment:
Although not related to your changes, it would be better to have some sort of await or callback from products_api.put_product_by_name instead of an arbitrary sleep period (something to keep in mind for future changes/improvements)

Reply:
Attempted a solution to this. Agree the arbitrary wait and no check isn't great; not sure if anything is gained from using `await` as there is no further process that can take place until the scope is definitely on the product, unless I have misunderstood `await`.

Feel free to make further suggestions on the amendments.